modules/scalacheck/shared/src/main/scala/weaver/scalacheck/CheckConfig.scala

```diff
@@ -3,104 +3,37 @@ package scalacheck
 
 import org.scalacheck.rng.Seed
 
-final class CheckConfig(
```

Comment:
you still need the `unapply` override in the companion object, which signals to the compiler that the traditional case class extractor should be eluded. Otherwise adding fields changes the shape of the generated unapply in non bincompat ways.